test/solvers/advanced/forward_backward_test.py

```diff
@@ -0,0 +1,181 @@
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Test for the forward-backward solver."""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+
+# External
+import pytest
+import numpy as np
+
+# Internal
+import odl
+from odl.solvers import forward_backward_pd
+from odl.util.testutils import all_almost_equal, example_element
+
+# Places for the accepted error when comparing results
+PLACES = 8
+
+
+def test_forward_backward_input_handling():
+    """Test to see that input is handled correctly."""
+
+    space1 = odl.uniform_discr(0, 1, 10)
+
+    lin_ops = [odl.ZeroOperator(space1), odl.ZeroOperator(space1)]
+    prox_cc_g = [odl.solvers.proximal_zero(space1),  # Identity operator
+                 odl.solvers.proximal_zero(space1)]  # Identity operator
+    prox_f = odl.solvers.proximal_zero(space1)  # Identity operator
+    grad_h = odl.ZeroOperator(space1)
+
+    # Check that the algorithm runs. With the above operators, the algorithm
+    # returns the input.
+    x0 = example_element(space1)
+    x = x0.copy()
+    niter = 3
+
+    forward_backward_pd(x, prox_f, prox_cc_g, lin_ops, grad_h, tau=1.0,
+                        sigma=[1.0, 1.0], niter=niter)
+
+    assert x == x0
+
+    # Testing that sizes needs to agree:
+    # To few sigma_i:s
+    with pytest.raises(ValueError):
+        forward_backward_pd(x, prox_f, prox_cc_g, lin_ops, grad_h, tau=1.0,
+                            sigma=[1.0], niter=niter)
+
+    # To many operators
+    prox_cc_g_to_many = [odl.solvers.proximal_zero(space1),
+                         odl.solvers.proximal_zero(space1),
+                         odl.solvers.proximal_zero(space1)]
+    with pytest.raises(ValueError):
+        forward_backward_pd(x, prox_f, prox_cc_g_to_many, lin_ops, grad_h,
+                            tau=1.0, sigma=[1.0, 1.0], niter=niter)
+
+    # Test for correct space
+    space2 = odl.uniform_discr(1, 2, 10)
+    x = example_element(space2)
+    with pytest.raises(ValueError):
+        forward_backward_pd(x, prox_f, prox_cc_g, lin_ops, grad_h, tau=1.0,
+                            sigma=[1.0, 1.0], niter=niter)
+
+
+def test_forward_backward_basic():
+    """Test for the forward-backward solver by minimizing ||x||_2^2. The
+    general problem is of the form
```

Comment:
If you insist on having a docstring instead of comments, you need to follow the docstring convention :-). One-liner first, then the rest.